tests/unit/contentproviders/test_zenodo.py

```diff
@@ -24,13 +24,47 @@ def test_detect():
     with patch.object(Zenodo, "_urlopen") as fake_urlopen:
         fake_urlopen.return_value.url = "https://zenodo.org/record/3232985"
         # valid Zenodo DOIs trigger this content provider
-        assert Zenodo().detect("10.5281/zenodo.3232985") == {"record": "3232985"}
-        assert Zenodo().detect("https://doi.org/10.5281/zenodo.3232985") == {
-            "record": "3232985"
+        assert Zenodo().detect("10.5281/zenodo.3232985") == {
+            "host": {
+                "hostname": ["https://zenodo.org/record/", "http://zenodo.org/record/"],
+                "api": "https://zenodo.org/api/records/",
+                "filepath": "files",
+                "filename": "filename",
+                "download": "links.download",
+                "type": "metadata.upload_type",
+            },
+            "record": "3232985",
         }
-        assert Zenodo().detect("https://zenodo.org/record/3232985") == {
-            "record": "3232985"
+        assert (
+            Zenodo().detect("https://doi.org/10.5281/zenodo.3232985")["record"]
+            == "3232985"
+        )
+        assert (
+            Zenodo().detect("https://zenodo.org/record/3232985")["record"] == "3232985"
+        )
+
+        # only two of the three calls above have to resolve a DOI
+        assert fake_urlopen.call_count == 2
+
+    with patch.object(Zenodo, "_urlopen") as fake_urlopen:
```

Comment:
I'd make this a new test method named something like `test_caltech_instance` or some such. This will help people from the future when part of the test breaks. I even wonder if we could combine the two tests into one where the input and expected output is passed in as parameters (https://docs.pytest.org/en/latest/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions)